core/domain/wipeout_service.py

```diff
@@ -339,18 +347,160 @@ def _pseudonymize_models(activity_related_models, pseudonymized_user_id):
+
+    suggestion_model_class = suggestion_models.GeneralSuggestionModel
+    general_suggestion_models = suggestion_model_class.query(ndb.OR(
+        suggestion_model_class.author_id == user_id,
+        suggestion_model_class.final_reviewer_id == user_id
+    )).fetch()
+    feedback_ids |= set([model.id for model in general_suggestion_models])
+
+    # The activity_mappings field might have only been partially generated, so
+    # we fill in the missing part for this activity category.
+    if models.NAMES.feedback not in pending_deletion_request.activity_mappings:
+        pending_deletion_request.activity_mappings[models.NAMES.feedback] = (
+            _generate_activity_to_pseudonymized_ids_mapping(feedback_ids)
+        )
+        save_pending_deletion_request(pending_deletion_request)
+
+    def _pseudonymize_models(feedback_related_models, pseudonymized_user_id):
+        """Pseudonymize user ID fields in the models.
+
+        Args:
+            feedback_related_models: list(BaseModel). Models whose user IDs
+                should be pseudonymized.
+            pseudonymized_user_id: str. New pseudonymized user ID to be used for
+                the models.
+        """
+        feedback_thread_models = [
+            model for model in feedback_related_models
+            if isinstance(model, feedback_thread_model_class)]
+        for feedback_thread_model in feedback_thread_models:
+            if feedback_thread_model.original_author_id == user_id:
+                feedback_thread_model.original_author_id = pseudonymized_user_id
+            if feedback_thread_model.last_nonempty_message_author_id == user_id:
+                feedback_thread_model.last_nonempty_message_author_id = (
+                    pseudonymized_user_id)
+        feedback_thread_model_class.put_multi(feedback_thread_models)
+
+        feedback_message_models = [
+            model for model in feedback_related_models
+            if isinstance(model, feedback_message_model_class)]
+        for feedback_message_model in feedback_message_models:
+            feedback_message_model.author_id = pseudonymized_user_id
+        feedback_message_model_class.put_multi(feedback_message_models)
+
+        general_suggestion_models = [
+            model for model in feedback_related_models
+            if isinstance(model, suggestion_model_class)]
+        for general_suggestion_model in general_suggestion_models:
+            if general_suggestion_model.author_id == user_id:
+                general_suggestion_model.author_id = pseudonymized_user_id
+            if general_suggestion_model.final_reviewer_id == user_id:
+                general_suggestion_model.final_reviewer_id = (
+                    pseudonymized_user_id)
+        feedback_thread_model_class.put_multi(general_suggestion_models)
+
+    feedback_mappings = (
+        pending_deletion_request.activity_mappings[models.NAMES.feedback])
```

Comment:
Should we be making these keys more specific? It seems a bit weird to me to combine all the different types of models within a single module, and then try to split them out again. It also seems a bit fragile because the location of the model may subsequently change during a codebase refactor (e.g. I can imagine suggestions becoming more of its own thing as the contributor dashboard develops).

Reply:
As I explained in the previous comment, all three models here are tied together and if we want to preserve the same pid across them they need to be in the same category.

I think that wipeout is hard to implement in a way that would always preserve its correct function when some changes to storage models are done. Any changes to storage models will need to be reviewed and thought about in the context of wipeout and if they change stuff related to users wipeout will need to be modified.

Reply:
Got it, thanks -- adding comments as mentioned above would help with conveying this understanding.